src/include/function/cast/functions/cast_string_to_functions.h

```diff
@@ -0,0 +1,290 @@
+#pragma once
+
+#include <cassert>
+
+#include "cast_string_non_nested_functions.h"
+#include "common/copier_config/copier_config.h"
+#include "common/exception/runtime.h"
+#include "common/string_format.h"
+#include "common/type_utils.h"
+#include "common/types/blob.h"
+#include "common/vector/value_vector.h"
+
+namespace kuzu {
+namespace function {
+
+struct CastStringToTypes {
+    static void copyStringToVector(common::ValueVector* vector, uint64_t rowToAdd,
+        std::string_view strVal, const common::CSVReaderConfig* csvReaderConfig);
+
+    template<typename T>
+    static inline bool tryCast(const char* input, uint64_t len, T& result) {
+        // try cast for signed integer types
+        return trySimpleIntegerCast<T, true>(input, len, result);
+    }
+
+    template<typename T>
+    static inline void operation(
+        common::ku_string_t& input, T& result, common::ValueVector& /*resultVector*/) {
+        // base case: int64
+        simpleIntegerCast<T, true>((char*)input.getData(), input.len, result,
+            common::LogicalType{common::LogicalTypeID::INT64});
+    }
+
+    template<typename T>
+    static void operation(const char* input, uint64_t len, T& result,
```

Comment:
will combine into 1 function after implementing casting const char* to ku_string_t